src/uu/tr/src/operation.rs

```diff
@@ -91,18 +95,22 @@ pub enum Sequence {
 }
 
 impl Sequence {
-    pub fn flatten(&self) -> Box<dyn Iterator<Item = u8>> {
+    pub fn flatten_non_lower_upper(&self) -> Box<dyn Iterator<Item = Self>> {
```

Comment:
seem that you are duplicating a bunch of code with flatten? 
maybe it could be deduplicated a bit, no?

Reply:
Logicwise, this code does not duplicate things. flatten_all calls flatten_non_upper_lower for everything that is not [:lower:]/[:upper:].